potassium/potassium.py

```diff
@@ -219,14 +225,22 @@ def handle(path):
 
         @flask_app.route('/__status__', methods=["GET"])
         def status():
+            idle_time = 0
+            inference_time = int((time.time() - self._inference_start_time)*1000)
```

Comment:
shouldn't inference_time be 0 if there's no inference currently running? Right not if i'm reading this correct it will pretty much always be > 0

Reply:
worth noting that there's a minor race condition to watch out for when implementing this as well since there will be a period where either inference start time hasn't been updated or the lock hasn't been acquired yet

Reply:
Good question, depends if we want to calculate the cold boot into the inference time. Here I opted to include it with the reasoning that if someone (for some weird reason) has a cold boot which is very long or is in an infinite loop. The timeout would still kick in and kill the replica. So from a user-point-of-view. "I set this value in the UI to 15min" => "not possible to have a replica running longer than 15min and thus not getting billed for more than 15min".

Does this sound reasonable?

Reply:
offline discussion: let's move this to the be only around the inference itself and not the cold boot.

Reply:
Just noting here. Moving this away from the replica manager creates the following behaviour:
1. inference timeout is reached => ForceDestroy() replica
2. replica manager notices that the replica is being terminated and in EnsureNoDeadReplicas() it spawns a replacement for it

So if a user makes one request (and only one) that times out the timeline would be something like:
replica1 is spawned & booted => replica1 does inference => replica1 is terminated due to timeout => replica2 is spawned => replica2 is terminated due to idle timeout.

What has been the original idea with respawning replicas in the EnsureNoDeadReplicas() function? 🤔

Reply:
Yeah, and confirming that moving the idle timeout logic to the available state (away from the replica manager) causes an infinite loop 😅 idle timeout is hit => EnsureNoDeadReplicas() creates a new one. And this loops on and on.